service-workers/service-worker/navigation-preload/allow-read-after-respondwith.https.html

```diff
@@ -0,0 +1,62 @@
+<!DOCTYPE html>
+<title>Service Worker: Nav preload: Allow read after respondwith</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="../resources/test-helpers.sub.js"></script>
+<script>
+  const scope = 'resources/slow-doc.py';
+
+  async function cleanup() {
+    const reg = await navigator.serviceWorker.getRegistration(scope);
+    if (!reg) return;
+    if (reg.scope == new URL(scope, location).href) {
+      return reg.unregister()
+    };
+  }
+
+  function testResultMessage(testName) {
+    return new Promise(resolve => {
+      navigator.serviceWorker.addEventListener('message', function listener(event) {
+        if (event.data.testName != testName) return;
+        navigator.serviceWorker.removeEventListener('message', listener);
+        resolve(event.data.value);
+      });
+    });
+  }
+
+  function waitForActivated(sw) {
```

Comment:
Can you use the wait_for_state() helper function here?

https://github.com/w3c/web-platform-tests/blob/6dbe2c7a9ee4bee4e024ad2c479eb7fbc64d3dff/service-workers/service-worker/resources/test-helpers.sub.js#L79